### DISCUSSION TITLE
Best way to roll a JWT in Edge Function?

### DISCUSSION THREAD

**Discussion by Alexa-Green** [supabase/supabase#21374] [Questions] [answered]
Using Edge Functions for APNS and need to store a JWT and mint a new one and save it accordingly once that one is expired. It should work like the following:

1. When the edge function needing the JWT runs, the existing token is fetched from the db and checked for expiration time
2. If the token is expired, the edge function creates a new jwt and overwrites the old one in the db for later retrieval (but safely/securely/not visible to any clients)

Where/how can I store this in the db to be accessed from an edge function so it is stored safely and securely and not accessible to any clients? I see how to set secrets as env vars but need this to be happening from an edge function.

**Comment by GaryAustin1** [comment, marked as answer]
You could put your table in a custom schema and only allow service_role to access it.  Then your edge function would use service_role.  Technically though the API can access but only with service role key.

If that is still not acceptable you would need to use the custom schema and use a direct database connection as the postgres user to access the table from the edge function with a postgres library.

**Comment by Alexa-Green** [comment]
@GaryAustin1 Great, thank you! I did that and it is working. Do I need to allow postgres to access it too? I see some of the Edge Functions operate as Postgres user-- or at least maybe one initiated from a cron job?
